### PR TITLE
Hanging fix

### PR DESCRIPTION
**Pull request by fabpot**
Reverting the revert

### CODE REVIEW

**Comment by fabpot** [comment]
Tested on Linux and MacOS and it seems to behave correctly.

**Comment by fabpot** [comment]
Indeed, I've reverted most of #308 by @tucksaun.

**Comment by fabpot** [comment]
I've again tested this patch extensively and I don't see any issues. I might miss something, but I think this is pretty solid now. At least, much better than the current state.

File: local/runner.go [hunk 1]
```diff
@@ -254,36 +252,23 @@ func (r *Runner) Run() error {
...
-			case sig := <-sigChan:
-				logger.Info().Msgf(`Signal "%s" received, exiting`, sig)
-				return nil
-			case <-restartChan:
-				logger.Debug().Msg("Received restart")
-				timer.Stop()
-			case <-timer.C:
-				logger.Debug().Msg("Received timer message")
-			}
+			return nil
```
Comment (tucksaun) [inline local/runner.go:271]: I have the feeling that returning nil here means a command not setup to always restart (hence everything else than PHP SAPIs) will not restart when done. Am I correct?

**Pull request merged**